source/frontend/src/features/acquisition/hooks/useAcquisitionFileExport.ts

```diff
@@ -0,0 +1,52 @@
+import axios, { AxiosError } from 'axios';
+import fileDownload from 'js-file-download';
+import { useCallback } from 'react';
+import { useDispatch } from 'react-redux';
+import { hideLoading, showLoading } from 'react-redux-loading-bar';
+import { toast } from 'react-toastify';
+
+import * as actionTypes from '@/constants/actionTypes';
+import { catchAxiosError } from '@/customAxios';
+import {
+  IPaginateAcquisition,
+  useApiAcquisitionFile,
+} from '@/hooks/pims-api/useApiAcquisitionFile';
+import { IApiError } from '@/interfaces/IApiError';
+import { logRequest, logSuccess } from '@/store/slices/network/networkSlice';
+
+export const useAcquisitionFileExport = () => {
+  const { exportAcquisitionFiles: apiExportAcquisitionFiles } = useApiAcquisitionFile();
+  const dispatch = useDispatch();
+
+  const exportAcquisitionFiles = useCallback(
+    async (
+      filter: IPaginateAcquisition,
+      outputFormat: 'excel',
+      fileName = `Acquisition_File_Export.xlsx`,
+      requestId = 'properties-report',
+    ) => {
+      dispatch(logRequest(requestId));
+      dispatch(showLoading());
+      try {
+        const { data, status } = await apiExportAcquisitionFiles(filter, outputFormat);
+        dispatch(logSuccess({ name: requestId, status }));
+        dispatch(hideLoading());
+        // trigger file download in client browser
+        fileDownload(data, fileName);
+      } catch (e) {
+        if (axios.isAxiosError(e)) {
+          const axiosError = e as AxiosError<IApiError>;
+          if (axiosError?.response?.status === 409) {
+            toast.error('Export contains no data');
+          } else {
+            catchAxiosError(e, dispatch, actionTypes.DELETE_PARCEL);
+          }
+          dispatch(hideLoading());
+        }
```

Comment:
A couple issues I see here:
- I would question why are we triggering the download in the "use..." hook. I'd rather just return the raw blob data and let the consumer of the hook decide what to do with it. Please see Devin's PR for an example. I'm doing the same in my export PR. This hook should be concerned with calling the API and returning the response - not with also triggering the file save popup.
- I see this on line 42 `catchAxiosError(e, dispatch, actionTypes.DELETE_PARCEL);` - is this copy/paste? This hook is not related to parcels at all...

Reply:
One more comment. I also don't think this new functionality warrants a separate "use..." hook. The method **exportAcquisitionFiles** could be added to the **useAcquisitionProvide** hook. The functionality is sufficiently relevant to that provider that it should go there instead of a separate hook with just one function

Reply:
This is the patter used for exporting leases. Not the first time I get comments for using patterns used in other sections of the code.

Reply:
Sorry if my comments came in as a bit harsh - totally not my intention. As I see it, PR comments are more "suggestions" than anything (unless it is some glaring error in the code). In my view, separating the download from the useXXX hook makes it more re-usable. Also, leases is the oldest portion of the codebase so the patterns used there might have been refined lately